### PR TITLE
Checkstyle: increase severity of all existing rules #624

### LINKED ISSUE #624: Checkstyle: wrong position of curly brace not triggering error
In the #623 PR, I have accidentally adopt the [Allman style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style), which deviates from our coding standard, yet not triggering any checkstyle error

https://github.com/reposense/RepoSense/blob/0f6d4f305890442739f045c1c8eb5997c30bfaf2/src/main/java/reposense/parser/CsvParser.java#L55-L56

### PR DESCRIPTION
**Pull request by emer7**
Fix #624 

```
RepoSense has various coding style rules which are
enforced by Checkstyle for the Java codebase.

However, some of those rules only generate warning and not error,
which will still pass the CI and may cause some coding style violation
to slip through.

Let's fix this by set all the violations for the rules to error.
```

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -162,7 +162,6 @@
 
     <module name="LeftCurly">
```
Comment (myteo) [inline config/checkstyle/checkstyle.xml]: Can make it self close since its empty: `<module name="LeftCurly"/>` and move comments on top.

**Comment by emer7** [comment]
@reposense/devs any idea why the netlify fails?

**Comment by fzdy1914** [comment]
It's ok now.

**Comment by emer7** [comment]
@reposense/stage2-reviewers ready for review
I removed the severity for right curly brace as I think it is implicitly mentioned in coding style.
The rest I don't remove it, what do you think?

**Comment by eugenepeh** [comment]
>@reposense/stage2-reviewers ready for review
I removed the severity for right curly brace as I think it is implicitly mentioned in coding style.
The rest I don't remove it, what do you think?

Those that are to be strictly enforced should be removed, otherwise more of such related issues are just going to pop out. While the issue is only mentions the curly brace, it implies the fault with current checkstyle configuration and hence should be rectified as a whole. With that being said, the issue is only targeting on the severity, hence should still be a marginal job.

**Comment by emer7** [comment]
> > @reposense/stage2-reviewers ready for review
> > I removed the severity for right curly brace as I think it is implicitly mentioned in coding style.
> > The rest I don't remove it, what do you think?
> 
> Those that are to be strictly enforced should be removed, otherwise more of such related issues are just going to pop out. While the issue is only mentions the curly brace, it implies the fault with current checkstyle configuration and hence should be rectified as a whole. With that being said, the issue is only targeting on the severity, hence should still be a marginal job.

@eugenepeh I don't know which rules need to be enforced again though. Currently we have warning on Javadoc comment and Logger variable case (should be LOGGER)

**Comment by eugenepeh** [comment]
>@eugenepeh I don't know which rules need to be enforced again though. Currently we have warning on Javadoc comment and Logger variable case (should be LOGGER)

Can fix all the warnings and enforce all the rules.

**Comment by emer7** [comment]
> > @eugenepeh I don't know which rules need to be enforced again though. Currently we have warning on Javadoc comment and Logger variable case (should be LOGGER)
> 
> Can fix all the warnings and enforce all the rules.

@eugenepeh enforced all

File: src/main/java/reposense/RepoSense.java [hunk 2]
```diff
@@ -26,11 +26,18 @@
...
+ * The main RepoSense class
+ */
 public class RepoSense {
     private static final Logger logger = LogsManager.getLogger(RepoSense.class);
     private static final int SERVER_PORT_NUMBER = 9000;
     private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("E MMM d HH:mm:ss yyyy z");
 
+    /**
+     * The main method as the entry point of the program
+     * @param args
```
Comment (eugenepeh) [inline src/main/java/reposense/RepoSense.java]: no need `@param` if its not providing useful info

File: src/main/java/reposense/RepoSense.java [hunk 3]
```diff
@@ -26,11 +26,18 @@
...
+/**
+ * The main RepoSense class
+ */
 public class RepoSense {
     private static final Logger logger = LogsManager.getLogger(RepoSense.class);
     private static final int SERVER_PORT_NUMBER = 9000;
     private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("E MMM d HH:mm:ss yyyy z");
 
+    /**
+     * The main method as the entry point of the program
```
Comment (eugenepeh) [inline src/main/java/reposense/RepoSense.java]: Can just leave it as `The entry point of the program`

File: src/main/java/reposense/authorship/analyzer/AnnotatorAnalyzer.java [hunk 4]
```diff
@@ -45,6 +45,9 @@ public static void aggregateAnnotationAuthorInfo(FileInfo fileInfo, Map<String,
         }
     }
 
+    /**
+     * Find the author that authors a line
```
Comment (eugenepeh) [inline src/main/java/reposense/authorship/analyzer/AnnotatorAnalyzer.java]: `Extracts the author name from the given {@code line}, finds the corresponding {@code Author} in {@code authorAliasMap}, and returns the result.`

**Comment by emer7** [comment]
@eugenepeh Ready for review

File: src/main/java/reposense/git/GitClone.java [hunk 5]
```diff
@@ -62,6 +62,10 @@ public static void clone(RepoConfiguration repoConfig)
         }
     }
 
+    /**
+     * Clones a repo given the location and folder name.
+     * @throws IOException
```
Comment (eugenepeh) [inline src/main/java/reposense/git/GitClone.java]: Should include description when adding a tag. There is no point adding one, if you're not providing useful information, as its already included in the function header.

File: src/main/java/reposense/model/Author.java [hunk 6]
```diff
@@ -131,6 +131,9 @@ public PathMatcher getIgnoreGlobMatcher() {
         return ignoreGlobMatcher;
     }
 
+    /**
+     * Adds all ignore glob list to {@code ignoreGlobList}.
```
Comment (eugenepeh) [inline src/main/java/reposense/model/Author.java]: `all ignore glob list`? Also, there are 2 `ignoreGlobList`, need to be clearer.

File: src/main/java/reposense/model/AuthorConfiguration.java [hunk 7]
```diff
@@ -123,11 +123,17 @@ public static void propagateIgnoreGlobList(Author author, List<String> ignoreGlo
...
+    /**
+     * Adds author to the {@code AuthorList}.
+     */
     public void addAuthor(Author author) {
         authorList.add(author);
         setAuthorDetails(author);
     }
 
+    /**
+     * Overloads {@code addAuthor} and {@code propagateIgnoreGlobList} methods.
```
Comment (eugenepeh) [inline src/main/java/reposense/model/AuthorConfiguration.java]: Overload method should repeat the original method's header and include the new information.
The phrase `Overloads {@code addAuthor}` doesn't help the user when she reads this javadoc alone.

File: src/main/java/reposense/model/ConfigCliArguments.java [hunk 8]
```diff
@@ -47,6 +47,9 @@ public Path getAuthorConfigFilePath() {
         return authorConfigFilePath;
     }
 
+    /**
+     * Compares the equality of {@code ConfigCliArguments} objects.
+     */
     public boolean equals(Object other) {
```
Comment (eugenepeh) [inline src/main/java/reposense/model/ConfigCliArguments.java]: if you use the `@Override` tag you prob don't need javadocs

File: src/main/java/reposense/parser/AuthorConfigCsvParser.java [hunk 9]
```diff
@@ -10,6 +10,9 @@
 import reposense.model.AuthorConfiguration;
 import reposense.model.RepoLocation;
 
+/**
+ * Parses {@code author-config.csv} file.
```
Comment (eugenepeh) [inline src/main/java/reposense/parser/AuthorConfigCsvParser.java]: `Container for the values parsed from {@code author-config.csv} file.`

File: src/test/java/reposense/template/GitTestTemplate.java [hunk 10]
```diff
@@ -97,6 +103,9 @@ public FileResult getFileResult(String relativePath) {
         return FileInfoAnalyzer.analyzeFile(config, fileinfo);
     }
 
+    /**
+     * Asserts the correctness of file analysis.
```
Comment (eugenepeh) [inline src/test/java/reposense/template/GitTestTemplate.java]: `Asserts the correctness of file analysis with regards to the contribution made by author named in {@code FAKE_AUTHOR_NAME}.`

**Comment by emer7** [comment]
@reposense/stage2-reviewers ready for review

File: src/main/java/reposense/system/LogsManager.java [hunk 11]
```diff
@@ -108,6 +108,9 @@ private static FileHandler createFileHandler() throws IOException {
         return fileHandler;
     }
 
+    /**
+     * Creates a {@code ConsoleHandler} for console output.
```
Comment (eugenepeh) [inline src/main/java/reposense/system/LogsManager.java]: `Creates a {@code ConsoleHandler} to output the log to console.`

File: src/main/java/reposense/system/CommandRunner.java [hunk 12]
```diff
@@ -5,9 +5,14 @@
...
 import reposense.util.SystemUtil;
 
+/**
+ * Contains command running related functionalities.
+ */
 public class CommandRunner {
 
-
+    /**
+     * Runs a command.
```
Comment (eugenepeh) [inline src/main/java/reposense/system/CommandRunner.java]: `Spawns a backend terminal process, with working directory at {@code path}, to executes the {@code command}.`

File: src/main/java/reposense/report/ReportGenerator.java [hunk 13]
```diff
@@ -210,6 +213,9 @@ private static void generateIndividualRepoReport(
         FileUtil.writeJsonFile(authorshipSummary.getFileResults(), getIndividualAuthorshipPath(repoReportDirectory));
     }
 
+    /**
+    * Generates a report for empty repo.
```
Comment (eugenepeh) [inline src/main/java/reposense/report/ReportGenerator.java]: `Generates a report at the {@code repoReportDirectory}.`

File: src/main/java/reposense/report/RepoCloner.java [hunk 14]
```diff
@@ -143,6 +143,9 @@ private void cleanupPrevRepoFolder() {
         }
     }
 
+    /**
+    * Deletes a directory given the root path.
```
Comment (eugenepeh) [inline src/main/java/reposense/report/RepoCloner.java]: `Deletes the {@code root} directory.`

File: src/main/java/reposense/parser/CsvParser.java [hunk 15]
```diff
@@ -77,6 +80,9 @@ public CsvParser(Path csvFilePath) throws IOException {
         return results;
     }
 
+    /**
+     * Checks if there is malformed line.
```
Comment (eugenepeh) [inline src/main/java/reposense/parser/CsvParser.java]: `Checks if the {@code line} contains values at the mandatory positions in CSV format.`

File: src/main/java/reposense/model/AuthorConfiguration.java [hunk 16]
```diff
@@ -123,11 +123,19 @@ public static void propagateIgnoreGlobList(Author author, List<String> ignoreGlo
...
```
Comment (eugenepeh) [inline src/main/java/reposense/model/AuthorConfiguration.java]: Actually what I meant is like this, which is the common javadocs format for overload methods.
```
Adds {@code author} to the {@code AuthorList}, and propagates {@code ignoreGlobList} to the {@code author}.

**Comment by emer7** [comment]
@eugenepeh fixed

**Comment by emer7** [comment]
@yamidark @yong24s ready for review 🙏

File: src/main/java/reposense/RepoSense.java [hunk 17]
```diff
@@ -26,11 +26,17 @@
 import reposense.system.ReportServer;
 import reposense.util.FileUtil;
 
+/**
+ * The main RepoSense class
```
Comment (yamidark) [inline src/main/java/reposense/RepoSense.java]: Missing index `.` at end

File: src/main/java/reposense/git/exception/GitCloneException.java [hunk 18]
```diff
@@ -1,6 +1,8 @@
 package reposense.git.exception;
 
-
+/**
+ * Represents the exception of git clone.
```
Comment (yamidark) [inline src/main/java/reposense/git/exception/GitCloneException.java]: Can rephrase this to follow format of the other `Exception` (e.g. `Signals that a commit is not found` in `CommitNotFoundException`)

**Comment by yamidark** [comment]
@emer7 Do update the proposed merge message, as this PR is now increasing severity of many other rules besides the 1 currently mentioned.

**Comment by yamidark** [comment]
Slight nits to proposed message:
```
RepoSense has various coding style rules which are enforced by
Checkstyle for the Java codebase.

However, some of those rules only generate warning and not error,
which will still pass the CI and allow some coding style violation
to slip through.

Let's fix this by setting all the violations for the coding style
rules to error instead.
```
